AzureIotPnpDps.X/platform/cloud/wifi_service.c

```diff
@@ -218,7 +218,26 @@ static void wifiCallback(uint8_t msgType, void* pMsg)
 			}
 			else if (pstrWifiState->u8CurrState == M2M_WIFI_DISCONNECTED)
 			{
-				debug_printWarn("  WiFi: DISCONNECTED: %d", pstrWifiState->u8ErrCode);
+				switch (pstrWifiState->u8ErrCode)
+                {
+                    case M2M_ERR_SCAN_FAIL:
+                        debug_printError("  WiFi: DISCONNECTED. Failed to detect AP during scan. Check SSID & authType");
+                        break;
+                    case M2M_ERR_JOIN_FAIL:
+                        debug_printError("  WiFi: DISCONNECTED, Error Code: %d.  Check WiFi Credentials", pstrWifiState->u8ErrCode);
+                        break;
+                    case M2M_ERR_AUTH_FAIL:
+                        debug_printError("  WiFi: DISCONNECTED. Failed to authenticate. Check pass phrase");
+                        break;
+                    case M2M_ERR_ASSOC_FAIL:
+                        debug_printError("  WiFi: DISCONNECTED, Error Code: %d.  Check WiFi Credentials", pstrWifiState->u8ErrCode);
+                        break;
+                    case M2M_ERR_CONN_INPROGRESS:
+                        break;
+                    default:
+                        debug_printError("  WiFi: DISCONNECTED, Error Code: %d.  Check WiFi Credentials", pstrWifiState->u8ErrCode);
+                        break;
+                }
```

Comment:
```suggestion
                {
                    case M2M_ERR_SCAN_FAIL:
                        debug_printError("  WiFi: DISCONNECTED. Failed to detect AP during scan. Check SSID & authType");
                        break;
                    case M2M_ERR_AUTH_FAIL:
                        debug_printError("  WiFi: DISCONNECTED. Failed to authenticate. Check pass phrase");
                        break;
                    case M2M_ERR_CONN_INPROGRESS:
                        break;
                    case M2M_ERR_ASSOC_FAIL:
                    case M2M_ERR_JOIN_FAIL:
                    default:
                        debug_printError("  WiFi: DISCONNECTED, Error Code: %d.  Check WiFi Credentials", pstrWifiState->u8ErrCode);
                        break;
                }
```

Reply:
(group the cases which give the same behavior)

Reply:
Hi Dane. 

Thanks for your feedback.  I updated the case statement, grouping like error messages.  Please let me know if you have other suggestions.

Regards,

Jason